JavaScript/packages/datatypes-date-time/src/timexRangeResolver.js

```diff
@@ -146,13 +146,47 @@ const resolveByTimeRangeConstraints = function (candidates, timexConstraints) {
 
     const resolution = [];
     for (const timex of candidates) {
-        const r = resolveTime(new TimexProperty(timex), collapsedTimeRanges);
-        Array.prototype.push.apply(resolution, r);
+        const t = new TimexProperty(timex);
+        if (t.types.has('timerange')) {
+            const r = resolveTimeRange(t, collapsedTimeRanges);
+            Array.prototype.push.apply(resolution, r);
+        }
+        else if (t.types.has('time')) {
+            const r = resolveTime(t, collapsedTimeRanges);
+            Array.prototype.push.apply(resolution, r);
+        }
     }
 
     return removeDuplicates(resolution);
 };
 
+const resolveTimeRange = function (timex, constraints) {
+
+    const candidate = timexHelpers.timeRangeFromTimex(timex);
+
+    const result = [];
+    for (const constraint of constraints) {
+
+        if (timexConstraintsHelper.isOverlapping(candidate, constraint)) {
+
+            const start = Math.max(candidate.start.getTime(), constraint.start.getTime());
+            const time = new Time(start);
+
+            const resolved = new TimexProperty(timex.timex);
```

Comment:
The clone is probably better (this actually clones via a string - albeit a small string but still its a little weak.) I need to do a little work across the board on copying/cloning it would probably simplify the code.